docs/src/Modal.doc.js

```diff
@@ -528,6 +528,72 @@ function Example(props) {
   />
 );
 
+card(
+  <Example
+    name="Outside Event Exception Test"
+    description={`
+      Testing that outside event is working as expected with portals
+    `}
+    defaultCode={`
+function Example(props) {
+  const [showModal, setShowModal] = React.useState(false);
+  const [hasLoaded, setHasLoaded] = React.useState(false);
+  const [showFlyout, setShowFlyout] = React.useState(false);
+
+  const anchorRef = React.useRef();
+
+  const handleToggleModal = () => {
+    setShowModal(!showModal);
+    setHasLoaded(false);
+  }
+
+  return (
+    <Box marginLeft={-1} marginRight={-1}>
+      <Box padding={1}>
+        <Button
+          text="View Modal"
+          onClick={handleToggleModal}
+        />
+        {showModal && (
+          <Modal
+            accessibilityModalLabel="View flyout modal"
+            heading="Modal"
+            onDismiss={handleToggleModal}
+          >
+            <Box margin={4} ref={anchorRef}>
+              <Button text="Open Flyout" onClick={() => setShowFlyout(true)} />
```

Comment:
@emilyalwayss there seems to be a bug when I take the following steps:
1. Open the modal
2. Click the "Show flyout" button => Flyout shows correctly
3. Click outside the modal (closes the modal)
4. Re-open the modal
5. Click the "Show flyout" button => Flyout does not show up

Could you take a look and fix this bug?

Reply:
fixed! good catch